front/lib/resources/group_resource.ts

```diff
@@ -366,70 +368,207 @@ export class GroupResource extends BaseResource<GroupModel> {
     return groups.map((group) => new this(GroupModel, group.get()));
   }
 
-  async addMember(
+  async getActiveMembers(auth: Authenticator): Promise<UserResource[]> {
+    const owner = auth.getNonNullableWorkspace();
+    if (!this.activeMembers) {
+      const memberships = await GroupMembershipModel.findAll({
+        where: {
+          groupId: this.id,
+          workspaceId: owner.id,
+          startAt: { [Op.lte]: new Date() },
+          [Op.or]: [{ endAt: null }, { endAt: { [Op.gt]: new Date() } }],
+        },
+      });
+
+      this.activeMembers = await UserResource.listByModelIds(
+        memberships.map((m) => m.userId)
+      );
+    }
+    return this.activeMembers;
+  }
+
+  async addMembers(
     auth: Authenticator,
-    user: UserType,
+    users: UserType[],
     transaction?: Transaction
-  ): Promise<
-    Result<
-      undefined,
-      {
-        type:
-          | "user_not_found"
-          | "user_not_workspace_member"
-          | "group_not_regular"
-          | "user_already_group_member";
-      }
-    >
-  > {
+  ): Promise<Result<undefined, Error>> {
     // Checking that the user is a member of the workspace.
     const owner = auth.getNonNullableWorkspace();
-    const userResource = await UserResource.fetchById(user.sId);
-    if (!userResource) {
-      return new Err({ type: "user_not_found" });
+
+    if (users.length === 0) {
+      return new Ok(undefined);
     }
-    const workspaceMembership =
-      await MembershipResource.getActiveMembershipOfUserInWorkspace({
-        user: userResource,
-        workspace: owner,
-        transaction,
-      });
 
-    if (!workspaceMembership) {
-      return new Err({ type: "user_not_workspace_member" });
+    const userIds = users.map((u) => u.sId);
+    const userResources = await UserResource.listByIds(userIds);
+    if (userResources.length !== userIds.length) {
+      return new Err(
+        userIds.length === 1
+          ? new Error("The user was not found.")
+          : new Error("Some users were not found.")
+      );
+    }
+    const workspaceMemberships = await MembershipResource.getActiveMemberships({
+      users: userResources,
+      workspace: owner,
+      transaction,
+    });
+
+    if (
+      new Set(workspaceMemberships.map((m) => m.userId)).size !== userIds.length
+    ) {
+      return new Err(
+        userIds.length === 1
+          ? new Error("The user is not member of the workspace.")
+          : new Error("Some users are not member of the workspace.")
+      );
     }
 
     // Users can only be added to regular groups.
     if (this.type !== "regular") {
-      return new Err({ type: "group_not_regular" });
+      return new Err(new Error("Not a regular group, cannot be updated."));
     }
 
     // Check if the user is already a member of the group.
-    const existingMembership = await GroupMembershipModel.findOne({
-      where: {
+    const activeMembers = await this.getActiveMembers(auth);
```

Comment:
Do we really need the users here, can't we simply return the user ids?

Reply:
we actually need the sIds